database-connector/src/main/java/life/qbic/projectmanagement/persistence/repository/OpenbisConnector.java

```diff
@@ -131,32 +180,167 @@ public List<Analyte> retrieveAnalytes() {
         .map(Analyte::new).toList();
   }
 
+  /**
+   * Creates a reference to one or more {@link Sample}s in the data repository to connect project
+   * data. A project must be provided, a unique experiment is created with each batch.
+   *
+   * @param project the {@link Project} for which samples should be created
+   * @param sampleBatch the batch of {@link Sample}s to be created in the data repo
+   * @since 1.0.0
+   */
+  public void addBatch(Project project,
+      List<life.qbic.projectmanagement.domain.project.sample.Sample> sampleBatch) {
+    String projectCodeString = project.getProjectCode().value();
+    List<SampleCreation> samplesToRegister = new ArrayList<>();
+
+    String newExperimentCode = findFreeExperimentCode(projectCodeString);
+    createOpenbisExperiment(DEFAULT_SPACE_CODE, projectCodeString, newExperimentCode);
+    ExperimentIdentifier newExperimentID = new ExperimentIdentifier(DEFAULT_SPACE_CODE,
+        projectCodeString, newExperimentCode);
+
+    sampleBatch.forEach(sample -> {
+      SampleCreation sampleCreation = new SampleCreation();
+      sampleCreation.setCode(sample.sampleCode().code());
+      sampleCreation.setTypeId(new EntityTypePermId("Q_TEST_SAMPLE"));
+      sampleCreation.setSpaceId(new SpacePermId(DEFAULT_SPACE_CODE));
+      Map<String, String> props = new HashMap<>();
+
+      props.put("Q_SECONDARY_NAME", sample.biologicalReplicateId().toString());
+      props.put("Q_EXTERNALDB_ID", sample.label());
+      props.put("Q_SAMPLE_TYPE", getSampleTypeFromAnalysis(sample.analysisType()));
+
+      sampleCreation.setProperties(props);
+
+      sampleCreation.setExperimentId(newExperimentID);
+      samplesToRegister.add(sampleCreation);
+
+    });
+    try {
+      createOpenbisSamples(samplesToRegister);
+    } catch (Exception e) {
+      deleteOpenbisExperiment(newExperimentID);
+      throw e;
+    }
+  }
+
+  private String getSampleTypeFromAnalysis(Optional<String> analysisType) {
+    Map<String, String> analysisTypeToOpenBISType = new HashMap<>();
+    analysisTypeToOpenBISType.put("RNA-Seq", "RNA");
+    analysisTypeToOpenBISType.put("DNA-Seq", "DNA");
+    return analysisTypeToOpenBISType.get(analysisType.get());
```

Comment:
favoring a static lookup map, will be then extended soon with enums of the analysis types

Reply:
not addressed yet, right?